### PR TITLE
[1LP][RFR] Automate test hard reboot for azure

### PR DESCRIPTION
**Pull request by bsquizz**
{{ pytest: -v -k test_hard_reboot_unsupported --use-provider azure --long-running cfme/tests/cloud/test_instance_power_control.py }}

This is a negative test, hard reboot is not supported in azure and an error message should be thrown by CFME.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0e20f4f99b7d0bba4c8c6f0ab3b25eee1ade7a21:

`cfme/tests/cloud/test_instance_power_control.py`:
- :bangbang: Line 260:101: **E501** *line too long (101 > 100 characters)*
- :heavy_exclamation_mark: Line 262:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 271:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5ddc09b01bf8dc4413465942335a3aecb1a6df73:

`cfme/tests/cloud/test_instance_power_control.py`:
- :bangbang: Line 261:13: **E126** *continuation line over-indented for hanging indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/cloud/test_instance_power_control.py [hunk 1]
```diff
@@ -248,6 +248,32 @@ def test_hard_reboot(appliance, provider, testing_instance, verify_vm_running, s
     soft_assert(provider.mgmt.is_vm_running(testing_instance.name), "instance is not running")
 
 
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(AzureProvider))
```
Comment (quarckster) [inline cfme/tests/cloud/test_instance_power_control.py]: **REQUIRED**
```python
@pytest.mark.providers([AzureProvider])
```

File: cfme/tests/cloud/test_instance_power_control.py [hunk 2]
```diff
@@ -248,6 +248,32 @@ def test_hard_reboot(appliance, provider, testing_instance, verify_vm_running, s
...
+    # Message we are looking for is...
+    #   "Reset does not apply to at least one of the selected items"
+    # ...but we leave out 'Reset' in case that text changes in the future
+    txt = "does not apply to at least one of the selected items"
+    for message in view.flash.messages:
+        if txt in message.read():
+            error_msg_found = True
+
+    assert error_msg_found, ("Expected to see error containing text '{}' after "
+                             "attempting unsupported hard reboot".format(txt))
```
Comment (quarckster) [inline cfme/tests/cloud/test_instance_power_control.py]: **REQUIRED**
```python
view.flash.assert_message("Reset does not apply to at least one of the selected items")
```

File: cfme/tests/cloud/test_instance_power_control.py [hunk 3]
```diff
@@ -248,6 +248,20 @@ def test_hard_reboot(appliance, provider, testing_instance, verify_vm_running, s
     soft_assert(provider.mgmt.is_vm_running(testing_instance.name), "instance is not running")
 
 
+@pytest.mark.providers([AzureProvider])
```
Comment (quarckster) [inline cfme/tests/cloud/test_instance_power_control.py]: **REQUIRED**
I'm sorry, it should be:
```python
@pytest.mark.provider([AzureProvider])
```
My bad.

File: cfme/tests/cloud/test_instance_power_control.py [hunk 4]
```diff
@@ -248,6 +248,20 @@ def test_hard_reboot(appliance, provider, testing_instance, verify_vm_running, s
...
+def test_hard_reboot_unsupported(appliance, provider, testing_instance):
```
Comment (quarckster) [inline cfme/tests/cloud/test_instance_power_control.py]: **REQUIRED**
`appliance` and `provider` fixtures are not needed.

**Comment by bsquizz** [comment]
@quarckster `@pytest.mark.provider([AzureProvider])` and my guess is that it may be because the provider marker is already used under the root pytestmark? I am switching back to using the `uncollectif` method as this is the convention used by all the other tests in this file at the moment, but let me know if you have other ideas. :)

**Pull request merged**